src/v/archival/ntp_archiver_service.cc

```diff
@@ -1243,6 +1243,7 @@ ss::future<ntp_archiver_upload_result> ntp_archiver::upload_segment(
   upload_candidate candidate,
   std::vector<ss::rwlock::holder> segment_read_locks,
   std::optional<std::reference_wrapper<retry_chain_node>> source_rtc) {
+    std::ignore = segment_read_locks; // prevent unused arg warning
```

Comment:
is this what we settled on? things that seem better:

1. `std::vector<ss::rwlock::holder> /*segment_read_locks*/`
2. `[[maybe_unused]] std::vector<ss::rwlock::holder> segment_read_locks`
3. `auto holder = std::move(segment_read_locks)`

Reply:
Went with option 3